stdlib/REPL/src/REPLCompletions.jl

```diff
@@ -387,14 +387,17 @@ get_value(sym::QuoteNode, fn) = isdefined(fn, sym.value) ? (getfield(fn, sym.val
 get_value(sym::GlobalRef, fn) = get_value(sym.name, sym.mod)
 get_value(sym, fn) = (sym, true)
 
-# Return the value of a getfield call expression
-function get_value_getfield(ex::Expr, fn)
-    # Example :((top(getfield))(Base,:max))
-    val, found = get_value_getfield(ex.args[2],fn) #Look up Base in Main and returns the module
-    (found && length(ex.args) >= 3) || return (nothing, false)
-    return get_value_getfield(ex.args[3], val) #Look up max in Base and returns the function if found.
-end
-get_value_getfield(sym, fn) = get_value(sym, fn)
+# Return the type of a getfield call expression
+function get_type_getfield(ex::Expr, fn::Module)
+    length(ex.args) == 3 || return Any, false # should never happen, but just for safety
+    obj, x = ex.args[2:3]
+    objt, found = get_type_getfield(obj, fn)
+    found || return Any, false
+    fld = isa(x, QuoteNode) ? x.value : x
```

Comment:
If `x` isn't a QuoteNode, then it is some expression, not a Symbol
```suggestion
    if x isa QuoteNode
        fld = x.value
    elseif isexpr(x, :quote) || isexpr(x, :inert)
        fld = x.args[1]
    else
        fld = nothing # we don't know how to get the value of variable `x` here
    end
    fld isa Symbol || return Any, false
```